src/Domain/ElementDistribution.hpp

```diff
@@ -0,0 +1,71 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <cstddef>
+#include <utility>
+#include <vector>
+
+#include "Domain/Structure/ElementId.hpp"
+
+namespace domain {
+
+/*!
```

Comment:
[picking line at random] should/could there be a reference that describes the algorithm a bit more? If not that's fine since I think Wikipedia actually does a decent job :shrug: